### ISSUE TITLE
Pressing `Ctrl +` in the main text area crashes Zim

### ISSUE THREAD

**Issue by llogick** [zim-desktop-wiki/zim-desktop-wiki#2421]
**Describe the bug**
Zim gets terminated with a SIGSEGV after pressing `Ctrl +` in the main text area

**To Reproduce**
Start Zim
Press `Ctrl` and `+`

**Expected behavior**
I have a habit of pressing `Ctrl +` to zoom in, I didn't actually expect anything, but would be nice :)

**Screenshots**
NA

**Environment (please complete the following information):**
- 0.75.2 from KDE` Discover
- Linux
- English

**Additional context**
Cursor has to be in the text area (what would be a "Page" ?)

For issues triggered by specific page content, please add an (anonymized) example that allows reproducing this issue.
NA

For issues with no visible output / error, check the debug log for additional logging. This is available via the menu or by running "zim -D" from the commandline.
```
❯ zim -D
DEBUG: Loading config from: <ConfigFile: /home/user/.config/zim/preferences.conf>
DEBUG: New extendable: <zim.plugins.InsertedObjectTypeMap object at 0x7fa930f66650>
DEBUG: Loading plugin: pageindex
DEBUG: Loading plugin: pathbar
DEBUG: Loading plugin: toolbar
DEBUG: Loading plugin: insertsymbol
DEBUG: Loading plugin: printtobrowser
DEBUG: Loading plugin: versioncontrol
INFO: This is zim 0.75.2
DEBUG: Python version is sys.version_info(major=3, minor=11, micro=4, releaselevel='final', serial=0)
DEBUG: Platform is posix
DEBUG: Not running from a source dir
DEBUG: Set XDG_DATA_HOME to /home/user/.local/share
DEBUG: Set XDG_DATA_DIRS to [<LocalFolder: /home/user/.local/share/flatpak/exports/share>, <LocalFolder: /var/lib/flatpak/exports/share>, <LocalFolder: /usr/local/share>, <LocalFolder: /usr/share>]
DEBUG: Set XDG_CONFIG_HOME to /home/user/.config
DEBUG: Set XDG_CONFIG_DIRS to [<LocalFolder: /home/user/.config/kdedefaults>, <LocalFolder: /etc/xdg>, <LocalFolder: /usr/local/etc/xdg>, <LocalFolder: /usr/etc/xdg>]
DEBUG: Set XDG_CACHE_HOME to /home/user/.cache
DEBUG: Connecting to /run/user/1000/zim-0.75.2-e63203e4
DEBUG: Got error in dispatch: Connection failed
DEBUG: Starting primary process
DEBUG: Start listening on: /run/user/1000/zim-0.75.2-e63203e4
DEBUG: Loading config from: <zim.notebook.info.VirtualFile object at 0x7fa927ff1c90>
DEBUG: Loading config from: /home/user/Notebooks/Notes/notebook.zim
DEBUG: Loading config from: /home/user/Notebooks/Notes/notebook.zim
DEBUG: Connecting to database file: /home/user/.cache/zim/notebook-home_user_Notebooks_Notes/index.db
DEBUG: Loading config from: /home/user/.cache/zim/notebook-home_user_Notebooks_Notes/state.conf
DEBUG: New extendable: <Notebook: Notes>
DEBUG: Load extension: <class 'zim.plugins.versioncontrol.VersionControlNotebookExtension'>
INFO: No VCS detected
DEBUG: Loading config from: <ConfigFile: /home/user/.config/zim/style.conf>
DEBUG: Autosave interval: 15 - use threads: True
INFO: Page changed on disk: Misc
INFO: Open page: Misc (Misc)
DEBUG: New extendable: <notebookview.NotebookView object at 0x7fa92c354580 (zim+gui+notebookview+NotebookView at 0x558a601d2860)>
DEBUG: Load extension: <class 'zim.plugins.insertsymbol.InsertSymbolPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.pageindex.PageIndexNotebookViewExtension'>
DEBUG: Action: toggle_panes(False)
DEBUG: Action: toggle_panes(True)
DEBUG: Load extension: <class 'zim.plugins.printtobrowser.PrintToBrowserPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.toolbar.ToolBarMainWindowExtension'>
DEBUG: New extendable: <mainwindow.MainWindow object at 0x7fa927ff0700 (zim+gui+mainwindow+MainWindow at 0x558a60192120)>
DEBUG: Load extension: <class 'zim.plugins.pathbar.PathBarMainWindowExtension'>
DEBUG: Load extension: <class 'zim.plugins.versioncontrol.VersionControlMainWindowExtension'>
DEBUG: Accelmap: /home/user/.config/zim/accelmap
DEBUG: Add window: MainWindow
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: Action: zoom_in
DEBUG: Switching to custom font implicitly because of zoom action
fish: Job 1, 'zim -D' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by introt** [comment]
`Ctrl` + `+` should indeed "zoom in" via enlarging the font; this seems to fail as per the debug log. Which Linux distribution are you running?

**Comment by introt** [comment]
As a workaround, check that Preferences > Interface > Use a custom font is enabled and points to an existing font

**Comment by llogick** [comment]
> `Ctrl` + `+` should indeed "zoom in" via enlarging the font; this seems to fail as per the debug log. Which Linux distribution are you running?

openSUSE Tumbleweed

> As a workaround, check that Preferences > Interface > Use a custom font is enabled and points to an existing font

This is a satisfactory solution, thank you !

**Comment by Joe-0237** [comment]
I experience this as well, `ctrl +` and `ctrl -` crash **Zim 0.75.1**
I'm on GNOME 44.3 on Fedora 38

**Comment by Joe-0237** [comment]
IDK if op played with the use custom font checkbox before experiencing this bug like I did. May be related.
Checking the box at `Edit > Preferences > Interface > Use a custom font` will allow zooming to work without crashing Zim as Ithink @introt was getting at but i did not correctly understand.

**Comment by mikro2nd** [comment]
Could not reproduce this. (Zim 0.75.2, KDE Plasma 5.18.8). I *do* have a custom font set.

However, while `ctrl +` worked fine, `ctrl -` did not zoom out -- had to use the menu to reset the font size.

**Comment by magicalskye** [comment]
i have this issue too, on a fresh arch install in fluxbox. SOMETIMES it works and sometimes it does not.

**Comment by robertvazan** [comment]
This can get quite annoying if you accidentally press `Ctrl -`, which shrinks the text and then you are unable to zoom back in, because `Ctrl +` crashes. I run Zim in flatpak on Gnome, Fedora 37. The workaround with custom font works.

**Comment by chromevanadium** [comment]
Just confirming on Fedora 38 on Wayland with the Gnome Desktop and an Intel® UHD Graphics (CML GT2) with the default dnf packaged Zim version 0.75.2.

Also selecting View - Zoom In/Out will crash Zim. I also attempted removing all Zim related configuration files in the home directory, but the issue persists

Fortunately,  the workaround of selecting a custom font works very well.

**Comment by abc-mikey** [comment]
Also was getting the same on Fedora 38 Wayland default package for 0.75.2. In my tests happened 100% of the time. 

Can also confirm that work around (setting Preferences > Interface > Use a custom font) works, but it could still do with a fix.

**Comment by LOLZpersonok** [comment]
I experienced this crash as well, but on macOS 14.0, using Zim 0.75.2 (264) on Apple Silicon. This crash also occurs when trying to use the zoom controls from the View menu in the menu bar. [Here are the logs on Pastebin](https://pastebin.com/qK2KBA7y) (they are quite long).
